src/algebraic_geometry/presheafed_space.lean

```diff
@@ -113,13 +113,32 @@ instance category_of_PresheafedSpaces : category (PresheafedSpace C) :=
 { hom := hom,
   id := id,
   comp := λ X Y Z f g, comp f g,
-  id_comp' := λ X Y f, by { ext1,
-    { rw comp_c, erw eq_to_hom_map, simp, apply comp_id }, apply id_comp },
-  comp_id' := λ X Y f, by { ext1,
-    { rw comp_c, erw congr_hom (presheaf.id_pushforward _) f.c,
-      simp, erw eq_to_hom_trans_assoc, simp }, apply comp_id },
-  assoc' := λ W X Y Z f g h, by { ext1,
-    repeat {rw comp_c}, simpa, refl } }
+  id_comp' := λ X Y f, begin
+    ext1,
+    { rw comp_c,
+      erw eq_to_hom_map,
+      simp only [eq_to_hom_refl, assoc, whisker_right_id'],
+      erw [comp_id, comp_id] },
+    apply id_comp
+  end,
+  comp_id' := λ X Y f, begin
+    ext1,
+    { rw comp_c,
+      erw congr_hom (presheaf.id_pushforward _) f.c,
+      simp only [comp_id, functor.id_map, eq_to_hom_refl, assoc, whisker_right_id'],
+      erw eq_to_hom_trans_assoc,
+      simp only [id_comp, eq_to_hom_refl],
+      erw comp_id },
+    apply comp_id
+  end,
+  assoc' := λ W X Y Z f g h, begin
+    ext1,
+    repeat {rw comp_c},
+    simp only [eq_to_hom_refl, assoc, functor.map_comp, whisker_right_id'],
+    erw comp_id,
+    congr,
```

Comment:
Can you briefly explain why these proofs got this much longer?

Reply:
Its because the previous one had non-terminal simps.
The only modification nessecary is probably an `erw comp_id` in each of the proofs.